rules-reviewed/technology-usage/connect-technology-usage.windup.xml

```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0"?>
+<ruleset id="technology-usage-http" xmlns="http://windup.jboss.org/schema/jboss-ruleset" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides statistical summaries of the http items that were found during the analysis.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,2.7.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.7.0.Final" />
+        </dependencies>
+        <phase>PostMigrationRulesPhase</phase>
+    </metadata>
+    <rules>
+        <rule id="technology-usage-connect-01000">
+            <when>
+                <graph-query discriminator="RMIServiceModel" />
+            </when>
+            <perform>
+                <technology-identified name="RMI">
+                    <tag name="Connect"/>
+                    <tag name="Other"/>
+                    <tag name="Java EE"/>
+                </technology-identified>
+            </perform>
+        </rule>
+
+        <rule id="technology-usage-connect-01100">
+            <when>
+                <or>
+                    <javaclass references="java.lang.System.{systemNativeMethods}({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <javaclass references="java.lang.Runtime.load({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <javaclass references="java.lang.Runtime.loadLibrary({*})">
+                        <location>METHOD_CALL</location>
+                    </javaclass>
+                    <javaclass references="com.sun.jna.{*}">
+                        <location>CONSTRUCTOR_CALL</location>
+                        <location>IMPORT</location>
+                        <location>INHERITANCE</location>
+                        <location>METHOD_CALL</location>
+                        <location>VARIABLE_DECLARATION</location>
+                    </javaclass>
+                </or>
+            </when>
+            <perform>
+                <technology-identified name="JNI">
+                    <tag name="Connect"/>
+                    <tag name="Other"/>
+                    <tag name="Java EE"/>
+                </technology-identified>
+            </perform>
+        </rule>
```

Comment:
Not a thing that directly apply only to this rule, but i'm asking myself if there would be a better way to handle this kind of rules without duplicating already existing rules.
For example, adding `<technology-tag level="INFORMATIONAL">JNI</technology-tag>` to [jni-native-code.windup.xml](https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/openshift/jni-native-code.windup.xml#L57) would make this rule without all the duplicated conditions in `<when>` section that would become:
```xml
            <when>
                <graph-query discriminator="TechnologyTagModel">
                    <property name="name">JNI</property>
                </graph-query>
            </when>
```

Reply:
To be honest, in this case I support @OndraZizka suggestion not to write new rules for identifying technologies, but rather update some existing rules and add identification there.

IMHO it would be much easier to maintain. Also it would reduce number of executed rules and possibly improve performance.

Reply:
Btw isn't `jni-native-code.windup.xml` executed only when cloud readiness rules are executed? 
(I don't know if that would be a problem or not.. ?)

Reply:
For completeness about your comment, you suggest to not write this `technology-usage-connect-01100` rule but rather add the `<technology-identified>` tag to [jni-native-code-00000](https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/openshift/jni-native-code.windup.xml#L39) in order to not duplicate the rule. Right?

Reply:
I think that approach might be better. But it was also part of discussion we had with Ondra. (He was suggesting to reuse classification model). Resolution was to stick to current model, so we can ignore that suggestion for now. 

But for future, if we ever get back to it again, I believe it would be better.

Reply:
It is already possible to reuse the classification information (or technology tags) in the when condition of these rules.